cupy/core/_routines_statistics.pyx

```diff
@@ -286,15 +280,26 @@ cdef ndarray _std(
     return _math._sqrt(ret, dtype=dtype, out=out)
 
 
+cdef _norm_preamble = '''
+template <typename T> __device__ T my_norm(T x) { return x * x; }
+
+__device__ float16 my_norm(const complex<float16>& x) { return norm(x); }
+__device__ float my_norm(const complex<float>& x) { return norm(x); }
+__device__ double my_norm(const complex<double>& x) { return norm(x); }
+'''
+
+
 cdef _var_core = ReductionKernel(
     'S x, T mean, T alpha', 'T out',
-    '(x - mean) * (x - mean)',
-    'a + b', 'out = alpha * a', '0', '_var_core')
+    'my_norm(static_cast<T>(x) - mean)',
```

Comment:
Is this optimization benefitial?
I think this kind of optimization is commonly done by the compiler.

Reply:
Sorry, never mind.
I overlooked complex variants use `std::norm`.